src/Transform.jl

```diff
@@ -40,17 +33,14 @@ function logit{NumType <: Number}(x::NumType)
 end
 
 
-function logit{NumType <: Number}(x::Array{NumType})
-    1.0 ./ (1.0 + exp(-x))
-end
-
-
 """
 Convert an (n - 1)-vector of real numbers to an n-vector on the simplex, where
 the last entry implicitly has the untransformed value 1.
 """
 function constrain_to_simplex{NumType <: Number}(x::Vector{NumType})
     if any(x .== Inf)
+        # Is there a bug here? If more than 1 entry in x is Inf, that just means
+        # that the last entry in z is 0, not the both entries in z are the same.
```

Comment:
I was proposing modifying `Optim.jl` to enforce box constraints itself, for the Newton trust region method, then no longer using doing the `vp_to_unconstrain` stuff (for box constrains, the simplex constraints are another matter). I think box constraints can be enforced by excluding out-of-bounds points from the trust region at every iteration. We've probably got more pressing things to work on now though...